### PR TITLE
bolts of resurrection kill the undead

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR makes bolts of resurrection instantly kill the undead (like how a bolt of death would instantly kill a non-undead creature) and makes bolts of death fully heal the undead (like how a bolt of resurrection would instantly heal a non-undead creature to full health). In addition, it makes some minor tweaks to some checks and things in the wands and bolts of resurrection and death (most notably, hellbound people can now be healed by magical bolts, but only if they're not dead (before, hellbound people would be entirely unaffected by bolts of healing, even if they weren't dead yet)).

## Why It's Good For The Game

It's a pretty common (T)RPG trope that healing items kill the undead, and that certain death/necromancy/negative energy spells heal the undead. We are still _/tg/_station, no?

Also, the people in this thread said that this was a good idea: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=24287&p=519846#p519846

## Changelog
:cl: ATHATH
balance: Bolts of resurrection now instantly kill the undead instead of fully healing them.
balance: Bolts of death now fully heal the undead instead of instantly killing them.
tweak: Hellbound people can now be healed by magical bolts, provided that they aren't dead (yet).
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
#41505
Should revert this PR with this in.

**Comment by ATH1909** [comment]
> #41505
> Should revert this PR with this in.

Eh, I feel like being dragged around (in a secure crate) by a holoparasite as a zombie wizard will still be kind of brokenly good, even if you're not being healed to full health every few seconds.

**Comment by HomocidlePancakes** [comment]
Well, “brokenly good” is literally wizard in a nutshell. If previously it was uncounterable, now it’s OP, which is average from the wizards standpoint.

>I feel like being dragged around (in a secure crate) 

I am dumb and don’t get what this means. Please elaborate.

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -18,7 +18,24 @@
...
+		if(isliving(M))
+			var/mob/living/L = M
+			if(L.mob_biotypes & MOB_UNDEAD) //negative energy heals the undead
+				if(L.hellbound && L.stat == DEAD)
+					return BULLET_ACT_BLOCK
+				if(iscarbon(L))
+					var/mob/living/carbon/C = L
+					C.regenerate_limbs()
+					C.regenerate_organs()
+				if(L.revive(full_heal = 1))
```
Comment (nemvar) [inline code/modules/projectiles/projectile/magic.dm]: revive with full_heal calls regenerate_limbs and regenerate_organ. full_heal is also a boolean.

**Comment by necromanceranne** [comment]
Could you add bible smacking harming undead unless it's the syndicate tome? That'd go really well with this change.

File: code/modules/projectiles/projectile/magic.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/projectiles/projectile/magic.dm]: oh, it does? cool, I'll remove the redundant limb and organ regeneration calls, then

**Comment by ATH1909** [comment]
> Could you add bible smacking harming undead unless it's the syndicate tome? That'd go really well with this change.

I'd like to put that in a future PR that'd also make bible smacking work on people with mechanical limbs (it's really weird that a chaplain of ratvar is unable to heal someone with bible smacks because of their "metallic taint"), if you don't mind. That is a good idea, though.

**Comment by davethwave** [comment]
Only problem I have with that is that bible smacking has been one of the only ways to heal skeletons right now.

**Pull request merged**